api-module-library/hubspot/tests/api.test.js

```diff
@@ -1,26 +1,376 @@
+            expect(response.results[0].properties).toHaveProperty('dealstage');
+        });
+    });
+
+    describe('HS Companies', () => {
+        let createRes;
+        beforeAll(async () => {
+            let body = {
+                domain: 'gitlab.com',
+                name: 'Gitlab',
+            };
+            createRes = await api.createCompany(body);
+        });
+
+        afterAll(async () => {
+            await api.archiveCompany(createRes.id);
+        });
+
+        it('should create a Company', async () => {
+            expect(createRes.properties.domain).toBe('gitlab.com');
+            expect(createRes.properties.name).toBe('Gitlab');
+        });
+
+        it('should return the company info', async () => {
+            let company_id = createRes.id;
+            let response = await api.getCompanyById(company_id);
+            expect(response.id).toBe(company_id);
+            // expect(response.properties.domain).to.eq('golabstech.com');
+            // expect(response.properties.name).to.eq('Golabs');
+        });
+
+        it.skip('should list Companies', async () => {
+            let response = await api.listCompanies();
+            expect(response.results[0]).toHaveProperty('id');
+            expect(response.results[0]).toHaveProperty('properties');
+            expect(response.results[0].properties).toHaveProperty('domain');
+            expect(response.results[0].properties).toHaveProperty('name');
+            expect(response.results[0].properties).toHaveProperty(
+                'hs_object_id'
+            );
+        });
+
+        it.skip('should update Company', async () => {
+            let body = {
+                name: 'Facebook 1',
+            };
+            let response = await api.updateCompany(
+                createRes.id,
+                body
+            );
+            expect(response.properties.name).toBe('Facebook 1');
+        });
+
+        it('should delete a company', async () => {
+            // Hope the after works!
         });
     });
 
-    afterAll(async () => {
-        await Manager.Credential.deleteMany();
-        await Manager.Entity.deleteMany();
-        await mongoose.disconnect();
+    describe.skip('HS Companies BATCH', () => {
+        let createResponse;
+        beforeAll(async () => {
+            let body = [
+                {
+                    properties: {
+                        domain: 'gitlab.com',
+                        name: 'Gitlab',
+                    },
+                },
+                {
+                    properties: {
+                        domain: 'facebook.com',
+                        name: 'Facebook',
+                    },
+                },
+            ];
+            createResponse = await api.createABatchCompanies(body);
+        });
+
+        afterAll(async () => {
+            createResponse.results.forEach(async (company) => {
+                await api.deleteCompany(company.id);
+            });
```

Comment:
Info: should we do it instead?
```suggestion
            return createResponse.results.map(async (company) => {
                return api.deleteCompany(company.id);
            });
```